### PR TITLE
Move print_usage() and print_version_information() out of controller

### PR DESCRIPTION
**Pull request by kmws**
c.f. #213 

Starting with low-hanging fruits. Comments welcome.

### CODE REVIEW

**Comment by Minoru** [comment]
No problem, that's the purpose of CI after all :)

**Comment by Minoru** [comment]
Looks like this one fallen through the cracks :) Reminding you about this PR in case you have a minute to finish it off.

**Comment by Minoru** [comment]
That file is auto-generated, and it just so happens that on Travis it's
generated after newsboat.cpp is compiledb (lucky us!). Run make depslist to
update mk/newsboat.mk, that'll make newsboat.cpp depend on xlicense.h,
which would hopefully fix the problem.

>

**Comment by kmws** [comment]
It helped locally, but there's still something wrong on Travis. :(

**Comment by Minoru** [comment]
Sorry, I didn't realize `make depslist` doesn't change `mk/newsboat.deps`. I'm not sure how it could work for you locally, but anyways—I added a corresponding line to `mk/newsboat.deps` now, and it should work now. Let's see what Travis says.

I'm not sure if I'll muster the energy to review this today, but I'll be sure to get to it tomorrow evening.

**Comment by Minoru** [comment]
Lol, I fell into exact same trap—it worked for me locally, but failed on Travis and it's obvious in hindsight that it *couldn't* work locally =\ Fingers crossed for the new commit.

**Comment by kmws** [comment]
Looks like clang doesn't like some includes? Will have a look tomorrow. Hope it's trivial. :)

**Comment by Minoru** [comment]
That last build failure is caused by my change, so I think it's on me to fix it :) I'll do it within the next two hours. And I still need to review this!

File: include/configparser.h [hunk 1]
```diff
@@ -9,7 +9,7 @@
 namespace newsboat {
 
 enum class action_handler_status {
-	OK = 0,
+	VALID = 0,
```
Comment (Minoru) [inline include/configparser.h]: Just curious: how did you find the clash? Did it cause any errors?

File: newsboat.cpp [hunk 2]
```diff
@@ -2,6 +2,7 @@
 #include <errno.h>
 #include <iostream>
 #include <ncurses.h>
+#include <sstream>
```
Comment (Minoru) [inline newsboat.cpp]: Out of curiosity: what's wrong about spamming `cout`? Given line buffering and how few lines we print out, I don't think this change is about performance. The only real difference I can see is that now messages generated by `print_usage` and `print_version_information` are guaranteed not to be interrupted by anything else, but I don't think that was the intention of your commit.

**Comment by Minoru** [comment]
Since I review commit-by-commit and some commits overlap, GitHub marked a few of my comments as outdated. Don't be fooled by that—all the comments still apply, even if the code I'm commenting have changed slightly :)

**Comment by Minoru** [comment]
Also, sorry about the delay. I'm bound to get better at time estimation any day now, given how many mistakes I make in that department!

File: include/configparser.h (hunk 1 above)
Comment (kmws) [reply include/configparser.h]: Yep, I think it was during adding ncurses.h to newsboat.cpp. I got a strange error and pinned it down to this include. Then I found out there's OK defined in ncurses.h (I think).

File: newsboat.cpp (hunk 2 above)
Comment (kmws) [reply newsboat.cpp]: To be honest it's just my personal preference. ;) I've always used streams to build up a string, but I'm ok with reverting this if you don't like it. Not gonna defend it very much.

**Comment by kmws** [comment]
>GitHub marked a few of my comments as outdated
Well, they are a couple of minutes old, so they are technically not new.... ;)

>sorry about the delay
It'd be stupid of me to grumble about the delay while it took me 21+ days to come back to this PR. :D

One more thing: I'd like to rename print_version_information() to just print_version(). Opinions?

File: newsboat.cpp (hunk 2 above)
Comment (Minoru) [reply newsboat.cpp]: Got it. Let's keep the change, then—I'm fine with it overall.

**Comment by Minoru** [comment]
In reply to an ancient (5 minutes old) comment by @kmws:

> One more thing: I'd like to rename print_version_information() to just print_version(). Opinions?

I'm in favour of the new name.

File: newsboat.cpp [hunk 3]
```diff
@@ -31,6 +104,11 @@ int main(int argc, char* argv[])
 	c.set_view(&v);
 	CLIArgsParser args(argc, argv);
 
+	if (args.should_print_usage) {
+		print_usage(args.program_name.c_str());
+		return EXIT_SUCCESS;
```
[the remark this answers is not in the document]
Comment (kmws) [reply newsboat.cpp]: I hope I got that right in my fix. If not, let me know.

File: test/cliargsparser.cpp [hunk 4]
```diff
@@ -208,7 +208,6 @@ TEST_CASE("Asks to print usage and exit with success if -h/--help is provided",
 
 		REQUIRE(args.should_print_usage);
 		REQUIRE(args.should_return);
-		REQUIRE(args.return_code == EXIT_SUCCESS);
```
Comment (Minoru) [inline test/cliargsparser.cpp]: Please restore this check as well, it's useful.

**Comment by Minoru** [comment]
It's DONE. Thank you very much for all the work, @kmws!

**Comment by kmws** [comment]
Thank you for the review and patience.

**Pull request merged**